kernel/thread.c

```diff
@@ -1053,6 +1032,26 @@ static long _run_thread(void* arg_)
             /* unmap any mapping made by the process */
             myst_release_process_mappings(process->pid);
 
+            if (process->exec_stack)
```

Comment:
Just to confirm, both the exec stack and crt data are setup by the process itself?

Reply:
Yes. The logic here is only for the main thread.